rclcpp/include/rclcpp/mapped_ring_buffer.hpp

```diff
@@ -196,6 +201,9 @@ class MappedRingBuffer : public MappedRingBufferBase
   bool
   has_key(uint64_t key)
   {
+    std::lock_guard<std::mutex> lock(data_mutex_);
+    bool did_replace = elements_[head_].in_use;
+    (void)did_replace;
```

Comment:
I see that you added the (void) line to avoid the compiler warning. But why not remove the variable if it is not used? Or does the above line have any side effects?